CHANGELOG.md

```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.4.0 (TBD)
+## 1.4.0 (2023-04-20)
 
 ### New Features
 
```

Comment:
There doesn't seem to be a way for me to add comments beside each new line so here is all of my feedback:

Each bullet should end with a period.

spark -> Spark

"to ease common day add/substract operations" -> "to make add and subtract operations easier"

"when creating UDF, UDTF and Stored Procedure" -> " when creating UDFs, UDTFs and stored procedures"

"to skip file uploads to stage in case the files are already present on stage." -> "to skip uploading files to a stage if the files are already on the stage."

"to take table name containing dots" -> "to take table names that contain dots"

"Dynamic Tables" -> "dynamic tables"

"in Private Preview" -> "in private preview"

"not supported in stored procedure yet" -> "not supported in stored procedures yet"

"Fixed a bug with `strtok_to_array` when a delimiter was passed it was throwing an exception"
->
"Fixed a bug in `strtok_to_array` where an exception was thrown when a delimiter was passed in."

"as select statement" -> "as select statements"

"Fixed a bug with `session.add_import` when the module has the same namespace with other dependencies" 
->
"Fixed a bug in `session.add_import` where the module had the same namespace as other dependencies."